### ISSUE TITLE
Add support for formatted address

### ISSUE THREAD

**Issue by Nyholm** [geocoder-php/Geocoder#558]
Google support returning formatted addresses. We should somehow support this.

**Comment by fyrye** [comment]
One suggestion to this would be to also allow the retrieval of responses from the providers, rather than just extracting the information in the `executeQuery` method.

Then a user would be able to extend a Provider object and instead of being forced to override the `executeQuery` method they could parse the response as desired with their own method.


For example:

```php
$adapter  = new \Http\Adapter\Guzzle6\Client();
$geocoder = new \Geocoder\Provider\GoogleMaps($adapter);

$geocoder->geocode(...);
$jsonResponse = $geocoder->getRepsonse(); //formatted address is available here
```

```php
public function geocode($address)
{
   $query = $address;
   //...
   return $this->parseData($this->executeQuery($query));
}

public function getResponse()
{
    return $this->response;
}

private function parseData($content)
{
    //...
   return $this->resultResults($results);
}

protected function executeQuery($query)
{
    $query = $this->buildQuery($query);
    if (!$this->response = json_encode((string) $this->getAdapter()->get($query)->getBody())) {
        throw new NoResult(sprintf('Could not execute query "%s".', $query));
    }
    //... move parsing of data out of executeQuery

    return $this->response;
}
```

**Comment by Nyholm** [comment]
Closed by #655
